sqlmesh/core/snapshot/evaluator.py

```diff
@@ -390,12 +390,25 @@ def _create_snapshot(
             is_dev=is_dev,
         )
 
+        evaluation_strategy = _evaluation_strategy(snapshot, self.adapter)
+
         with self.adapter.transaction(TransactionType.DDL), self.adapter.session():
             self.adapter.execute(snapshot.model.render_pre_statements(**render_kwargs))
 
-            _evaluation_strategy(snapshot, self.adapter).create(
-                snapshot.model, table_name, **render_kwargs
-            )
+            if is_dev and snapshot.is_materialized and self.adapter.SUPPORTS_CLONING:
+                tmp_table_name = f"{target_table_name}__schema_migration_source"
+                source_table_name = snapshot.table_name()
+
+                logger.info(f"Cloning table '{source_table_name}' into '{target_table_name}'")
+
+                evaluation_strategy.create(snapshot.model, tmp_table_name, **render_kwargs)
+                try:
+                    self.adapter.clone_table(target_table_name, snapshot.table_name(), replace=True)
+                    self.adapter.alter_table(target_table_name, tmp_table_name)
```

Comment:
This reads to me like this:
- Create clone from prod version using the final dev version table name
- Alter the dev version table name (so the clone we just made) to have a temp table name
- Drop temp table name

I think I'm not following this right.

Reply:
Sorry never mind got it now. Was thinking rename for some reason instead of alter meaning change the schema itself.